src/tests/tracing/eventpipe/applystartuphook/ApplyStartupHookValidation.cs

```diff
@@ -64,6 +66,90 @@ public static async Task<bool> TEST_ApplyStartupHookAtStartupSuspension()
+            Logger.logger.Log($"Server name is '{serverName}'");
+            Task<bool> subprocessTask = Utils.RunSubprocess(
+                currentAssembly: Assembly.GetExecutingAssembly(),
+                environment: new Dictionary<string,string> 
+                {
+                    { Utils.DiagnosticPortsEnvKey, serverName }
+                },
+                duringExecution: async (pid) =>
+                {
+                    ReverseServer server = new ReverseServer(serverName);
+                    Logger.logger.Log("Waiting to accept diagnostic connection.");
+                    using (Stream stream = await server.AcceptAsync())
+                    {
+                        Logger.logger.Log("Accepted diagnostic connection.");
+
+                        IpcAdvertise advertise = IpcAdvertise.Parse(stream);
+                        Logger.logger.Log($"IpcAdvertise: {advertise}");
+
+                        SessionConfiguration config = new(
+                            circularBufferSizeMB: 1000,
+                            format: EventPipeSerializationFormat.NetTrace,
+                            providers: new List<Provider> { 
+                                new Provider(AppEventSource.SourceName, 0, EventLevel.Verbose)
+                            });
+
+                        Logger.logger.Log("Starting EventPipeSession over standard connection");
+                        using Stream eventStream = EventPipeClient.CollectTracing(pid, config, out ulong sessionId);
+                        Logger.logger.Log($"Started EventPipeSession over standard connection with session id: 0x{sessionId:X}");
+
+                        using EventPipeEventSource source = new(eventStream);
+                        TaskCompletionSource completionSource = new(TaskCreationOptions.RunContinuationsAsynchronously);
+
+                        source.Dynamic.All += (TraceEvent traceEvent) =>
+                        {
+                            if (AppEventSource.SourceName.Equals(traceEvent.ProviderName) && nameof(AppEventSource.Running).Equals(traceEvent.EventName))
+                                completionSource.TrySetResult();
+                        };
+
+                        _ = Task.Run(() => source.Process());
+
+                        Logger.logger.Log($"Send ResumeRuntime Diagnostics IPC Command");
+                        // send ResumeRuntime command (0x04=ProcessCommandSet, 0x01=ResumeRuntime commandid)
+                        var message = new IpcMessage(0x04,0x01);
+                        Logger.logger.Log($"Sent: {message.ToString()}");
+                        IpcMessage response = IpcClient.SendMessage(stream, message);
+                        Logger.logger.Log($"received: {response.ToString()}");
+                        CheckResponse(response, ref fSuccess);
+                        
+                        Logger.logger.Log("Start waiting for any managed event.");
```

Comment:
Is it fair to say any event? the tcs only gets a result on `AppEventSource.Running` being emitted.

Reply:
"Start waiting for any event that indicates managed code is running."?